backend/src/openai.ts

```diff
@@ -302,97 +298,6 @@ async function chatGptChatCompletion(
 	}
 }
 
-// estimate the tokens on a single completion
-function estimateMessageTokens(
-	message: ChatCompletionMessageParam | null | undefined
-) {
-	if (message) {
-		if ((message as ChatCompletionMessage).tool_calls) {
-			const funcMessage = message as ChatCompletionMessage;
-			const toolCalls = funcMessage.tool_calls?.map(
-				(toolCall) => toolCall.function
-			);
-			if (toolCalls) {
-				let estLTokens = 0;
-				for (const toolCallContent of toolCalls) {
-					// tool call not yet implemented in openai-chat-tokens so pass in as a function call
-					estLTokens += promptTokensEstimate({
-						messages: [
-							{
-								role: 'assistant',
-								function_call: toolCallContent,
-							} as ChatCompletionMessageParam,
-						],
-					});
-					return estLTokens;
-				}
-			}
-		} else {
-			const estTokens = promptTokensEstimate({ messages: [message] });
-			console.debug('message:', message, 'tokens:', estTokens);
-			return estTokens;
-		}
-	}
-	return 0;
-}
-
-// take only the chat history to send to GPT that is within the max tokens
-function filterChatHistoryByMaxTokens(
-	chatHistory: ChatCompletionMessageParam[],
-	maxNumTokens: number
-): ChatCompletionMessageParam[] {
-	// estimate total prompt tokens including chat history and function call definitions
-
-	const estimatedTokens =
-		promptTokensEstimate({
-			messages: chatHistory,
-			functions: chatGptTools.map((tool) => tool.function),
-		}) + 5; // there is an offset of 5 between openai completion prompt_tokens
-	// if the estimated tokens is less than the max tokens, no need to filter
-
-	if (estimatedTokens <= maxNumTokens) {
-		return chatHistory;
-	}
-
-	console.log(
-		'Filtering chat history to fit inside context window. estimated_tokens = ',
-		estimatedTokens
-	);
-	let sumTokens = 0;
-	const filteredList: ChatCompletionMessageParam[] = [];
-
-	// reverse list to add from most recent
-	const reverseList = chatHistory.slice().reverse();
-
-	// always add the most recent message to start of list
-	filteredList.push(reverseList[0]);
-	sumTokens += estimateMessageTokens(reverseList[0]);
-
-	// if the first message is a system role add it to list
-	if (chatHistory[0].role === 'system') {
-		sumTokens += estimateMessageTokens(chatHistory[0]);
-		filteredList.push(chatHistory[0]);
-	}
-
-	// add elements after first message until max tokens reached
-	for (let i = 1; i < reverseList.length; i++) {
-		const element = reverseList[i];
-		const numTokens = estimateMessageTokens(element);
-		// if we reach end and system role is there skip as it's already been added
-		if (element.role === 'system') {
-			continue;
-		}
-		if (sumTokens + numTokens <= maxNumTokens) {
-			filteredList.splice(i, 0, element);
-			sumTokens += numTokens;
-		} else {
-			console.debug('Max tokens reached on completion = ', element);
-			break;
-		}
-	}
-	return filteredList.reverse();
-}
-
 // take only the completions to send to GPT
```

Comment:
Not really in PR scope, but I think this comment and the one on line 306 don't need to be there (I think they don't give any info beyond what's in the method name)